### ISSUE TITLE
bean-web crashes on prices that are zero.

### ISSUE THREAD

**Issue by blais** [beancount/beancount#157]
**[Original report](https://bitbucket.org/blais/beancount/issue/157) by Andrew Stein (Bitbucket: [AndrewStein9](https://bitbucket.org/AndrewStein9)).**

----------------------------------------

I have some prices that are zero in the beancount file. For example:
```
2011-09-14 price MCMXX                                 0 MCMXX
```

`bean-web` crashes when processing these.

This is the stack trace
```
Traceback (most recent call last):
  File "/usr/local/bin/bottle.py", line 862, in _handle
    return route.call(**args)
  File "/usr/local/bin/bottle.py", line 1740, in wrapper
    rv = callback(*a, **ka)
  File "/usr/local/lib/python3.6/site-packages/beancount-2.0b13-py3.6-macosx-10.12-x86_64.egg/beancount/web/web.py", line 1030, in wrapper
    app.price_map = prices.build_price_map(entries)
  File "/usr/local/lib/python3.6/site-packages/beancount-2.0b13-py3.6-macosx-10.12-x86_64.egg/beancount/ops/prices.py", line 116, in build_price_map
    for (date, rate) in remove_list]
  File "/usr/local/lib/python3.6/site-packages/beancount-2.0b13-py3.6-macosx-10.12-x86_64.egg/beancount/ops/prices.py", line 116, in <listcomp>
    for (date, rate) in remove_list]
decimal.DivisionByZero: [<class 'decimal.DivisionByZero'>]
127.0.0.1 - - [29/Jan/2017 16:43:05] "GET / HTTP/1.1" 500 741
```

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Looks like a bug, but that would trigger only in a rare case. 
Could you shave as much of your life as possible in order to reproduce this and share a minimal failing file?

**Comment by blais** [comment]
**Original comment by Andrew Stein (Bitbucket: [AndrewStein9](https://bitbucket.org/AndrewStein9)).**

----------------------------------------

Hi Martin, It took me a little while, but here is an example.

**Comment by blais** [comment]
**Original comment by Andrew Stein (Bitbucket: [AndrewStein9](https://bitbucket.org/AndrewStein9)).**

----------------------------------------

Hi Martin, looking again, I see that I had a bug where I define a price with the commodity in terms of itself -- a bad idea. So feel free to ignore this...

And perhaps have a check that there are no lines of the type

```
#!python

YYYY-MM-DD price XXX NNN XXX
```

or maybe just allow NNN to be 1.0

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Anything that crashes is a bug and I need to fix it
Thanks for reporting
Sorry there will be delays
Dealing with real life for another month or so open source dev will have to
wait

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #157: Building a price map with zero prices should not fail.
